README.md

```diff
@@ -1,17 +1,98 @@
+
+## Graph constraints
+
+Currently, the only supported graph constraint is `max_out_degree`, which limits how many outgoing relationships a source account can have.
+The constraint is specified during graph instantiation, and enforced during all later graph operations.
+
+## Graph operations
+
+Each account graph instance is a shared Move object, so anyone can interact with it.
+However, all graph operations are implicitly performed against the transaction `sender` account.
+
+### Manage relationships
+
+An account can manage its own outgoing relationships in a graph through these move calls:
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::add_relationship` - adds a relationship from the transaction `sender` account to an arbitrary target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::remove_relationship` - removes a relationship from the transaction `sender` account to an arbitrary target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::clear_relationship` - clears all relationships from the transaction `sender` account.
+
+### Manage account properties
+
+An account can optionally manage its own account properties in a graph.
+The type of account properties is specified by the graph creator during instantiation.
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::set_account_props` - sets account properties for the transaction `sender` account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::unset_account_props` - unsets account properties for the transaction `sender` account.
+
+### Manage relationship properties
+
+An account can optionally manage properties of its outgoing relationships in a graph.
+The type of relationship properties is specified by the graph creator during instantiation.
+
+- `<ACCOUNT_GRAPH_PKG>::account_graph::set_relationship_props` - sets properties for the relationship from the transaction `sender` account to a target account.
+- `<ACCOUNT_GRAPH_PKG>::account_graph::unset_relationship_props` - unsets properties for the relationship from the transaction `sender` account to a target account.
+
+## Bullshark Quests beneficiary graph
+
+The account graph has been adopted by the [Bullshark Quests](https://quests.mystenlabs.com/) as the official and only way to link a user's managed in-app wallet to their self-custody wallet.
+This enables applications that leverage managed wallets for their users to participate in Bullshark Quests, while allowing their users' Bullshark NFTs to remain in their self-custody wallets.
+
+This is achieved through the _beneficiary graph_ - an instance of account graph, where each relationship represents a _beneficiary designation_, between the _benefactor_ (managed wallet) and the _beneficiary_ (self-custody wallet).
+All user activities happening on the _benefactor_ address are automatically attributed to the _beneficiary_ address when Mysten Labs calculates user points at each interval.
```

Comment:
I think this is better and sufficiently vague about how Mysten may implement it.